### PR TITLE
Labels appregistry and operator.bundle are mutually exclusive

### PR DESCRIPTION
**Pull request by MartinBasti**
Introduce pre plugin reponsible for checking user settings. We want to
fail early if Labels are used incorrectly for operator builds instead of
waiting till post phase.

* OSBS-8634

Signed-off-by: Martin Bašti <mbasti@redhat.com>

Maintainers will complete the following section:
- [ ] Commit messages are descriptive enough
- [ ] "Signed-off-by:" line is present in each commit
- [ ] Code coverage from testing does not decrease and new code is covered
- [ ] JSON/YAML configuration changes are updated in the relevant schema
- [ ] Changes to metadata also update the documentation for the metadata
- [ ] Pull request includes link to an osbs-docs PR for user documentation updates
- [ ] New feature can be disabled from a configuration file

### CODE REVIEW

**Comment by MartinBasti** [comment]
counterpart https://github.com/containerbuildsystem/osbs-client/pull/940

**Comment by MartinBasti** [comment]
OMG, fixing py2 syntax error

```
E     File "/home/travis/build/containerbuildsystem/atomic-reactor/tests/plugins/test_check_user_settings.py", line 30
1988E       data = '\n'.join([base, *extra_labels, cmd])
1989E                               ^
1990E   SyntaxError: invalid syntax
```

File: tests/plugins/test_check_user_settings.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        source={"provider": "git", "uri": "asd"}
+    )
+    workflow.source = StubSource()
+    builder = StubInsideBuilder().for_workflow(workflow)
+    builder.set_df_path(str(tmpdir))
+    builder.tasker = flexmock()
+    workflow.builder = flexmock(builder)
+
+    if for_orchestrator:
+        workflow.buildstep_plugins_conf = [{'name': PLUGIN_CHECK_USER_SETTINGS}]
```
Comment (chmeliik) [inline tests/plugins/test_check_user_settings.py]: I might be missing something, but shouldn't this be `PLUGIN_BUILD_ORCHESTRATE_KEY`?

File: atomic_reactor/plugins/pre_check_user_settings.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    for incorrect options or mutually exclusive options
+    """
+    key = PLUGIN_CHECK_USER_SETTINGS
+    is_allowed_to_fail = False
+
+    def dockerfile_checks(self):
+        """Checks for Dockerfile"""
+        self.appregistry_bundle_label_mutually_exlucisve()
+
+    def appregistry_bundle_label_mutually_exlucisve(self):
```
Comment (chmeliik) [inline atomic_reactor/plugins/pre_check_user_settings.py]: \*exclusive

File: tests/plugins/test_check_user_settings.py (hunk 1 above)
Comment (MartinBasti) [reply tests/plugins/test_check_user_settings.py]: Hmm, you are right, I;m thinking if this section is even needed in test

Comment (MartinBasti) [reply tests/plugins/test_check_user_settings.py]: removed

File: atomic_reactor/plugins/pre_check_user_settings.py (hunk 2 above)
Comment (MartinBasti) [reply atomic_reactor/plugins/pre_check_user_settings.py]: fixed

**Pull request merged**